app/controllers/admin/assets_controller.rb

```diff
@@ -13,28 +13,30 @@ def index
     def unspam
       @asset.ham!
       @asset.update_attribute :is_spam, false
-      flash.notice = "Track was made public"
-      redirect_back(fallback_location: root_path)
     end
 
     def spam
       @asset.spam!
       @asset.update_attribute :is_spam, true
-      flash.notice = "Track was marked as spam"
-      redirect_back(fallback_location: root_path)
     end
 
+    # not used at the moment
     def mark_group_as_spam
       scope = Asset.where(permitted_params[:mark_spam_by])
       assets = scope.not_spam
 
       assets.map(&:spam!)
       assets.update_all(is_spam: true)
-      redirect_back(fallback_location: root_path)
+    end
+
+    def delete
+      Assets::SoftDeleteRelations.new(asset: @asset).execute
+      @asset.soft_delete
```

Comment:
This is non-blocking, as it's only going to apply to these 2 locations and console usage, but it's worth noting we'll never want to call these 2 things separately, as that would leave the Asset in a strange state — it would be better to have 1 method that we call here, similar to `User#soft_delete_with_relations` or `User.spam_and_mark_for_deletion` (or potentially using `SoftDeleteAssetWithRelations` as the command)

Reply:
Makes sense. i'll update.